pkg/controller/controller.go

```diff
@@ -93,6 +100,63 @@ func NewController(clientset kubernetes.Interface, ssclientset ssclientset.Inter
 			key, err := cache.MetaNamespaceKeyFunc(obj)
 			if err == nil {
 				queue.Add(key)
+				ns, _, err := cache.SplitMetaNamespaceKey(key)
+				if err != nil {
+					log.Printf("failed to get namespace and name from key: %v", err)
+					return
+				}
+				if _, ok := sInformers[ns]; !ok {
+					// chache.Indexers intitialization: https://github.com/kubernetes/client-go/blob/e7cd4ba474b5efc2882e377362c9aa8b407428d9/informers/core/v1/secret.go#L81
+					inf := corev1informers.NewFilteredSecretInformer(clientset, ns, 0, cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc}, tweakopts)
+					stopCh := make(chan struct{})
+
+					_, err = inf.AddEventHandler(cache.ResourceEventHandlerFuncs{
+						DeleteFunc: func(obj interface{}) {
```

Comment:
Inception! (add handled setting delete handler)

I would add a regular function here instead. It will be more readable.